tests/blackbox/testlib/dbus.py

```diff
@@ -0,0 +1,187 @@
+        }
+
+        pool_object_paths = [
+            path
+            for path, pool_obj in pool_objects.items()
+            if pool_obj["Name"] == pool_name
+        ]
+        if pool_object_paths == []:
+            return None
+
+        iface = dbus.Interface(
+            StratisDbus._BUS.get_object(StratisDbus._BUS_NAME, StratisDbus._TOP_OBJECT),
+            StratisDbus._MNGR_IFACE,
+        )
+        iface.DestroyPool(pool_object_paths[0])
+
+        return None
+
+    @staticmethod
+    def fs_list():
+        """
+        Query the file systems
+        :return: A dict,  Key being the fs name, the value being a dict with
+                          keys [POOL_NAME, USED_SIZE, UUID, SYM_LINK, CREATED,
+                                CREATED_TS]
+        """
+        objects = StratisDbus._get_managed_objects().items()
+
+        fs_objects = [
+            obj_data[StratisDbus._FS_IFACE]
+            for _, obj_data in objects
+            if StratisDbus._FS_IFACE in obj_data
+        ]
+
+        pool_path_to_name = {
+            obj: obj_data[StratisDbus._POOL_IFACE]["Name"]
+            for obj, obj_data in objects
+            if StratisDbus._POOL_IFACE in obj_data
+        }
+
+        result = {
+            fs_object["Name"]: pool_path_to_name[fs_object["Pool"]]
+            for fs_object in fs_objects
+        }
+
+        return result
+
+    @staticmethod
+    def fs_destroy(pool_name, fs_name):
+        """
+        Destroy a FS
+        :param pool_name:  Pool which contains the FS
+        :param fs_name: Name of FS to destroy
+        :return: None
+        """
+        objects = StratisDbus._get_managed_objects().items()
+
+        pool_objects = {
+            path: obj_data[StratisDbus._POOL_IFACE]
+            for path, obj_data in objects
+            if StratisDbus._POOL_IFACE in obj_data
+        }
+        fs_objects = {
+            path: obj_data[StratisDbus._FS_IFACE]
+            for path, obj_data in objects
+            if StratisDbus._FS_IFACE in obj_data
+        }
+
+        pool_object_paths = [
+            path
+            for path, pool_obj in pool_objects.items()
+            if pool_obj["Name"] == pool_name
+        ]
+        if pool_object_paths == []:
+            return None
+
+        fs_object_paths = [
+            path for path, fs_obj in fs_objects.items() if fs_obj["Name"] == fs_name
+        ]
+        if fs_object_paths == []:
+            return None
+
+        iface = dbus.Interface(
+            StratisDbus._BUS.get_object(StratisDbus._BUS_NAME, pool_object_paths[0]),
+            StratisDbus._POOL_IFACE,
+        )
+        iface.DestroyFilesystems(fs_object_paths)
+
+        return None
+
+    @staticmethod
+    def destroy_all():
+        """
+        Destroys all Stratis FS and pools!
+        :return: None
+        """
+        umount_mdv()
+
+        # Remove FS
+        for name, filesystem in StratisDbus.fs_list().items():
```

Comment:
```filesystem``` -> ```poolname``` and change line below also to match.

Reply:
Resolved.